### PR TITLE
Codestart modularize Dockerfiles

### PR DESCRIPTION
**Pull request by vsevel**
This is an alternate proposition for https://github.com/quarkusio/quarkus/pull/42589

there are 2 main `include`, which provide an overall view for docker files: 
 - `Dockerfile-jar.include.qute`
 - `Dockerfile-native.include.qute`

each one includes a `FROM` `include`, which will allow for easy override:
 - `Dockerfile-jar-from.include.qute`
 - `Dockerfile-native-from.include.qute`

all 4 templates `Dockerfile.tpl.qute...` merely include one of the 2 main `include`.

### CODE REVIEW

**Review by ia3andy — CHANGES_REQUESTED** [review]
You should be able to have only one include file for all types (native and non native), just use a Let https://quarkus.io/guides/qute-reference#let_section to set the value for jvm

**Review by ia3andy — APPROVED** [review]
Neat @vsevel

**Review by ia3andy — APPROVED** [review]
Perfect!

**Comment by maxandersen** [comment]
wonder how it relates to https://github.com/quarkusio/quarkus/pull/42316

**Comment by ia3andy** [comment]
@gsmet waiting for your final word on this one too to merge it

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.jvm
Comment (gsmet) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.jvm:2]: Here, I think we should use `fast-jar` and not `jvm` because that's really what differentiate the two in the context of this include: either you have a fast-jar or an uber-jar. The template file can stay as is. That's really about changing the `type` in this file and the one that is testing it.

**Review by gsmet — COMMENTED** [review]
Sorry for the delay. I like this approach a lot better, thanks.

There's only one thing that is bothering me a bit. Could you adjust? Then I think we can merge.

And we will have to discuss what you achieve here with the work of @iocanel on the dynamic generation of templates.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.jvm
Comment (gsmet) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles/base/src/main/docker/Dockerfile.tpl.qute.jvm:2]: @vsevel did you see this comment? It looks like this one is nearly ready to get merged?

**Review by maxandersen — COMMENTED** [review]
Yeah. Well want to figure how this relates to the regeneration work before we merge it. 

I think it can be aligned but as it is now we would end up with two approaches that are related but having duplicated maintanence and different level of customization.

**Comment by vsevel** [comment]
@maxandersen and @ia3andy what should we do?
I am ok if we close it as not planned, unless you see some value in this refactoring.

**Comment by vsevel** [comment]
after discussion, does not seem to bring enough value as this time,

**Pull request closed without merging**
